redis/src/main/java/org/hibernate/ogm/datastore/redis/impl/RedisConfiguration.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.redis.impl;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.hibernate.ogm.cfg.OgmProperties;
+import org.hibernate.ogm.cfg.impl.HostParser;
+import org.hibernate.ogm.cfg.spi.Hosts;
+import org.hibernate.ogm.datastore.redis.RedisProperties;
+import org.hibernate.ogm.util.configurationreader.impl.Validators;
+import org.hibernate.ogm.util.configurationreader.spi.ConfigurationPropertyReader;
+
+import com.lambdaworks.redis.RedisURI;
+
+/**
+ * @author Mark Paluch
+ */
+public class RedisConfiguration {
+
+	/**
+	 * The default host to connect to in case the {@link OgmProperties#HOST} property is not set
+	 */
+	private static final String DEFAULT_HOST = "localhost";
+
+	/**
+	 * The default timeout (5000ms) in case the {@link RedisProperties#TIMEOUT} property is not set
+	 */
+	private static final long DEFAULT_TIMEOUT = TimeUnit.MILLISECONDS.toMillis( 5000 );
+
+	private static final int DEFAULT_PORT = 6379;
+
+	private final Hosts hosts;
+	private final int databaseNumber;
+	private final String password;
+	private long timeout = DEFAULT_TIMEOUT;
+	private boolean ssl = false;
+
+
+	public RedisConfiguration(ConfigurationPropertyReader propertyReader) {
+		String host = propertyReader.property( OgmProperties.HOST, String.class )
+				.withDefault( DEFAULT_HOST )
+				.getValue();
+
+		Integer port = propertyReader.property( OgmProperties.PORT, Integer.class )
+				.withValidator( Validators.PORT )
+				.withDefault( null )
+				.getValue();
+
+		hosts = HostParser.parse( host, port, DEFAULT_PORT );
+
+		databaseNumber = propertyReader.property( OgmProperties.PORT, Integer.class )
```

Comment:
Is it right do initialize that with the port? A redis-specific parameter for this seems better suited.